src/metabase/lib/stage.cljc

```diff
@@ -414,3 +416,15 @@
     (cond-> query
       (and (lib.breakout/breakouts query) (lib.aggregation/aggregations query))
       append-stage)))
+
+(defmethod lib.query/can-run-method :mbql.stage/mbql
+  [query card-type]
+  (or (not= card-type :metric)
+      (let [aggregations (lib.aggregation/aggregations query 0)
+            breakouts    (lib.breakout/breakouts query 0)]
+        (and (= (lib.query/stage-count query) 1)
+             (= (count aggregations) 1)
+             (or (empty? breakouts)
+                 (and (= (count breakouts) 1)
+                      (-> (lib.breakout/breakout-column query (first breakouts))
```

Comment:
Note - `lib.breakout/breakout-column` doesn't change the `:effective-type` of the column when there is an extraction `:temporal-unit` on a breakout but `lib.metadata.calculation/metadata` does. We already depend on this, so I used the same function here and added tests for these cases.